azure_iot_edge/README.md

```diff
@@ -0,0 +1,146 @@
+# Agent Check: Azure IoT Edge
+
+## Overview
+
+[Azure IoT Edge][1] is a fully managed service to deploy Cloud workloads to run on Internet of Things (IoT) edge devices via standard containers.
+
+Use the Datadog-Azure IoT Edge integration to collect metrics and health status from IoT Edge devices.
+
+## Setup
+
+Follow the instructions below to install and configure this check for an IoT Edge device running on a device host.
+
+### Installation
+
+The Azure IoT Edge check is included in the [Datadog Agent][2] package.
+
+No additional installation is needed on your device.
+
+### Configuration
+
+It is recommended to configure the IoT Edge device so that the Agent runs as a custom module. Follow the official Microsoft documentation on [deploying Azure IoT Edge modules][3] for general information on installing and working with custom modules for Azure IoT Edge.
+
+Follow the steps below to configure the IoT Edge device, runtime modules, and the Datadog Agent to start collecting IoT Edge metrics.
+
+1. Edit your IoT Edge device `config.yaml` file:
+    - **Linux**: Make sure the `connect.management_uri` and `listen.management_uri` options point to a Unix Domain Socket (note that this is the default configuration). For example:
+
+        ```yaml
+        # /etc/iotedge/config.yaml
+
+        connect:
+            management_uri: "unix:///var/run/iotedge/mgmt.sock"
+            # ...
+
+        listen:
+            management_uri: "fd://iotedge.mgmt.socket"
+            # ...
+        ```
+
+    - **Windows**: Make sure the `connect.management_uri` and `listen.management_uri` options point to an HTTP endpoint. For example:
```

Comment:
No, on Windows that's not the default configuration. And I'm actually having issues with setting this up on Windows (whereas it does work quite fine on Linux - see our test setup).

The reason behind this diff is that we can't support connecting through a Unix Socket on Windows for now (due to a Python limitation: Python just hasn't implemented `AF_UNIX` on Windows yet). So we need a different strategy to get the status of the Security Manager on Windows.

Until now I thought that connecting through a regular TCP socket would do the trick. Unfortunately my testing yesterday shows that this is not trivial, as in the default installation IoT Edge ships with a stripped down version of Docker where eg `host.docker.internal` is not available, so I'm having issues setting `connect.management_uri` in a way that allows the Edge Agent container to talk to the Security Manager on the host. 🤷 

Other options I'm considering are:

- Drop the `security_manager.can_connect` service check - either on Windows only, or completely. Rationale: if the Edge Agent and Edge Hub are up and running, then that certainly means the Security Manager is fine too (since it's the one managing those two components).
- Find another way to check the Security Manager health on Windows, knowing that the Datadog Agent would have to reach to the host from a container. (But I'm not sure what that would be, any ideas? The Security Manager is a Windows service.)